hydra-main/src/main/java/com/addthis/hydra/job/spawn/Spawn.java

```diff
@@ -1656,6 +1668,7 @@ public boolean checkStatusForMove(String hostID) {
     }
 
     public boolean prepareTaskStatesForRebalance(Job job, JobTask task, boolean isMigration) {
+        // Why?
```

Comment:
I think it is conceivable that the task could have suddenly kicked in between the call to isInMovableState and the call to job.setTaskState .

If that were the case, the task state would show up as REBALANCE, but the true state would be RUNNING, which would be a bad outcome.

Holding the job lock here prevents the task from kicking until we have decided whether we are rebalancing this task.